SignalShareExtension/ShareViewController.swift

```diff
@@ -651,10 +647,23 @@ public class ShareViewController: UIViewController, ShareViewDelegate, SAEFailed
         }
     }
 
-    private func buildAttachment() -> Promise<SignalAttachment> {
```

Comment:
* I doubled-down on promises.  I think it makes this complex logic more readable.